abd-clam/src/core/dataset/vec2d.rs

```diff
@@ -50,6 +56,163 @@ impl<T: Send + Sync + Copy, U: Number> VecDataset<T, U> {
     }
 }
 
+impl<'a, V: Number, U: Number> VecDataset<&'a [V], U> {
+    /// Saves a numeric dataset to a specified path in binary format
+    ///
+    /// # Errors
+    /// Errors on file write issues or if a reordered index does not exist
+    pub fn save(&self, path: &Path) -> Result<(), String> {
+        let mut handle = File::create(path).map_err(|err| format!("{err}"))?;
+
+        // Write header (Basic protection against reading bad data)
+        handle.write_all(b"VEC2D").map_err(|err| format!("{err}"))?;
+
+        // Write dataset name
+        handle.write_all(self.name.as_bytes()).map_err(|err| format!("{err}"))?;
+
+        // Write null terminator
+        handle.write(&[0]).map_err(|err| format!("{err}"))?;
+
+        // Write cardinality
+        let cardinality = self.data.len();
+        let cardinality_bytes = cardinality.to_le_bytes();
+        handle.write_all(&cardinality_bytes).map_err(|err| format!("{err}"))?;
+
+        // Write dimensionality
+        let dimensionality = self.data[0].len();
+        let dimensionality_bytes = dimensionality.to_le_bytes();
+        handle
+            .write_all(&dimensionality_bytes)
+            .map_err(|err| format!("{err}"))?;
```

Comment:
Include the `type_name` after this. I just added the method to the `Number` trait. You'll have to update the dependency on `distances` to `1.4.1`.